cmd/api/http/handlers.go

```diff
@@ -122,6 +122,14 @@ func (h *BookHandler) createBook(w http.ResponseWriter, r *http.Request) {
 
 	storedBook, err := h.bookService.CreateBook(r.Context(), reqBook)
 	if err != nil {
+		if errors.Is(err, r.Context().Err()) {
+			errCtx := book.ErrResponse{
+				Code:    book.ErrResponseFromContext.Code,
+				Message: book.ErrResponseFromContext.Message + r.Context().Err().Error(),
+			}
+			responseJSON(w, http.StatusInternalServerError, errCtx)
```

Comment:
I've even tought off using `408 StatusRequestTimeout` , but it denotes an error from client, not from server:

> [15.5.9. ](https://www.rfc-editor.org/rfc/rfc9110.html#section-15.5.9)[408 Request Timeout](https://www.rfc-editor.org/rfc/rfc9110.html#name-408-request-timeout)
The 408 (Request Timeout) status code indicates that the server did not receive a complete request message within the time that it was prepared to wait.
If the client has an outstanding request in transit, it MAY repeat that request. If the current connection is not usable (e.g., as it would be in HTTP/1.1 because request delimitation is lost), a new connection will be used.

So, I think a 505 fits better here.

Reply:
504 is the best here. It's a gateway timeout, but it's widely used as a server timeout too.

Reply:
ok